warning still happens on Jetson TK1

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master ( bd786f3beab27e259b7c6d003a7700753e09e6c2 )
- Operating System / Platform => Ubuntu 14.04 Jetson TK1 (ARM 32bit)
- Compiler => GCC 4.8.4 + CUDA 6.5 (L4T 21.3)

##### Detailed description

 - during the compilation of photo module, following warning happens
```
/usr/local/cuda/include/cuda_runtime.h: In constructor ‘_GLOBAL__N__38_tmpxft_000068bd_00000000_6_nlm_cpp1_ii_5a9eff94::__cudaOccupancyB2DHelper::__cudaOccupancyB2DHelper(size_t)’:
/usr/local/cuda/include/cuda_runtime.h:1232:44: warning: declaration of ‘n’ shadows a member of 'this' [-Wshadow]
   inline __host__ CUDART_DEVICE __cudaOccupancyB2DHelper(size_t n) : n(n) {}
                                            ^
```
 - since it's from the cuda itself, I think it's better to suppress the warning
 - I'' create a PR to fix this issue

##### Steps to reproduce
 - checkout master
 - cmake and make as usual

--- a/modules/photo/CMakeLists.txt
+++ b/modules/photo/CMakeLists.txt
@@ -1,7 +1,7 @@
 set(the_description "Computational Photography")
 
 if(HAVE_CUDA)
-  ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Wmissing-declarations)
+  ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Wmissing-declarations -Wshadow)
 endif()
 
 ocv_define_module(photo opencv_imgproc OPTIONAL opencv_cudaarithm opencv_cudaimgproc WRAP java python)

